### DISCUSSION TITLE
The VCS integration should also lint commit messages and intervene with git hooks (maybe?)

### DISCUSSION THREAD

**Discussion by mkvlrn** [biomejs/biome#4817] [Ideas]
TL;DR: need biome to **KILL** those ~~two~~ three packages down there

DEAD packages obliterated by biome:

- ~~eslint~~ 💀 
- ~~bunch of eslint related crap~~ 💀
- ~~prettier~~ 💀
- ~~some prettier plugins (yeah, people actually installed those)~~ 💀
- ~~lint-staged~~ 💀 (thanks to VCS integration!)
- @commitlint/cli
- @commitlint/config-conventional
- husky 🐶 

Is there a possibility for biome, through VCS integration, to lint/verify the commit message and require the config-conventional/angular format? Our projects are looking leaner and leaner, and removing/reducing dev dependencies is part of an ongoing effort.

On top of that, how possible is for it to act as git hook, or maybe intervene with them? Would it be possible to have entries in `biome.json` under the VCS section that mirror git hooks, much like [husky](https://github.com/typicode/husky) does with plain files? This seems more far fetched than linting commit messages, but hey, maybe you guys can do it. And kill yet another package. **Dead**.

**Comment by Conaclos** [comment]
Hi!

It is currently out of scope of Biome, and it is not part of our roadmap.

For my projects I am using [validate-commit-message](https://www.npmjs.com/package/validate-commit-message). It is a dependency-free NPM package that checks if your commit messages satisfy the Angular commit convention.
It automatically set up a git hook for you.

If you need to manage hooks I could recommend [lefthook](https://www.npmjs.com/package/lefthook) that is dependency-free, fast, and nice to use.

**Reply by mkvlrn** [reply]
Aw, alright then.

But would it _really_ be out of Biome's scope?

> Biome is a performant toolchain for web projects, it aims to provide developer tools to maintain the health of said projects.

I would consider anything that is consistently in my `devDependencies`  fitting of that description. And I'm not trying to argue with you or start trouble! Just pointing out what I understand by scope. English is not my native language, and sometimes my foot goes in my mouth.

It's more about _not having to_ use extra tools than using other tools that could maybe reduce the dependency graph - the ones you recommended work, but so do the ones I'm currently using.

Thanks for checking it out anyway, hope this could be considered someday.
